packages/react-data-grid/src/Cell.tsx

```diff
@@ -1,131 +1,102 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { isElement, isValidElementType } from 'react-is';
 
-import { isFunction } from './common/utils';
+import { Column, CellMetaData, ExpandableOptions, RowData, SubRowOptions, FormatterProps, ColumnEventInfo } from './common/types';
 import { SimpleCellFormatter } from './formatters';
 import CellAction from './CellAction';
 import CellExpand from './CellExpander';
 import ChildRowDeleteButton from './ChildRowDeleteButton';
 import { isFrozen } from './ColumnUtils';
 
-const getSubRowOptions = ({ rowIdx, idx, rowData, expandableOptions: expandArgs }) => ({ rowIdx, idx, rowData, expandArgs });
-
-export default class Cell extends React.PureComponent {
-  static propTypes = {
-    rowIdx: PropTypes.number.isRequired,
-    idx: PropTypes.number.isRequired,
-    isSelected: PropTypes.bool,
-    wasPreviouslySelected: PropTypes.bool,
-    selectedColumn: PropTypes.object,
-    height: PropTypes.number,
-    column: PropTypes.object.isRequired,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.object, PropTypes.bool]),
-    isExpanded: PropTypes.bool,
-    isRowSelected: PropTypes.bool,
-    cellMetaData: PropTypes.object.isRequired,
-    handleDragStart: PropTypes.func,
-    className: PropTypes.string,
-    cellControls: PropTypes.any,
-    rowData: PropTypes.object.isRequired,
-    forceUpdate: PropTypes.bool,
-    expandableOptions: PropTypes.object.isRequired,
-    tooltip: PropTypes.string,
-    isScrolling: PropTypes.bool,
-    isCellValueChanging: PropTypes.func,
-    children: PropTypes.oneOfType([
-      PropTypes.arrayOf(PropTypes.node),
-      PropTypes.node
-    ]),
-    scrollLeft: PropTypes.number.isRequired
-  };
+const getSubRowOptions = ({ rowIdx, idx, rowData, expandableOptions: expandArgs }: Props): SubRowOptions => ({ rowIdx, idx, rowData, expandArgs });
+
+interface Props {
+  rowIdx: number;
+  idx: number;
+  height: number;
+  column: Column;
+  value: unknown;
+  isExpanded?: boolean;
+  isRowSelected?: object;
+  cellMetaData: CellMetaData;
+  handleDragStart(): void;
+  className?: string;
+  cellControls?: unknown;
+  rowData: RowData;
+  expandableOptions?: ExpandableOptions;
+  tooltip?: string;
+  isScrolling?: boolean;
+  scrollLeft: number;
+  lastFrozenColumnIndex?: number;
+}
 
+export default class Cell extends React.PureComponent<Props> {
   static defaultProps = {
     isExpanded: false,
-    value: '',
-    isCellValueChanging: (value, nextValue) => value !== nextValue
-  };
-
-  cell = React.createRef();
-
-  state = {
-    isCellValueChanging: false,
-    isLockChanging: false
+    value: ''
   };
 
-  componentWillReceiveProps(nextProps) {
```

Comment:
`componentWillReceiveProps` is not required as we can check the change in `componentDidUpdate`